lib/kickplan/concurrency.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Kickplan
+  module Concurrency
+    def self.included(klass)
+      klass.prepend Initializer
+    end
+
+    def memoize(name = caller_locations(1,1)[0].label, &block)
```

Comment:
The memoization variable defaults to the name of the caller function but can be overridden by an argument.